lib/workflows/email_shutoff.py

```diff
@@ -0,0 +1,117 @@
+from typing import List, Optional, Dict, Tuple
+from openstack_query.api.query_objects import ServerQuery
+from enums.query.props.server_properties import ServerProperties
+from enums.query.props.user_properties import UserProperties
+from structs.email.email_params import EmailParams
+from enums.query.query_presets import QueryPresetsGeneric
+from email_api.emailer import Emailer
+from structs.email.smtp_account import SMTPAccount
+
+
+def query_shutoff_vms(project_id: Optional[List[str]] = None):
+    """
+    Sends an email to a user about a shutoff (stopped) VM
+    :param project_id: optional field to provide a list of project IDs for specific projects,
+    otherwise query all projects
+    :returns: Dictionary with a list of VM names grouped by user ID
+    """
+
+    query_obj = ServerQuery()
+
+    query_obj.select(
+        ServerProperties.SERVER_NAME
+    ).where(
+        QueryPresetsGeneric.EQUAL_TO,
+        ServerProperties.SERVER_STATUS,
+        value="SHUTOFF")
```

Comment:
IMO: I think you should add an extra `where` clause to limit search to servers which haven't been updated in a while rather than all of them

```
.where(
    QueryPresetsDateTime.OLDER_THAN,
    ServerProperties.SERVER_LAST_UPDATED_DATE,
    days=int(days_threshold)
)
```
you can add `days_threshold` as an extra param to your workflow main function

Reply:
I would ask @jacob-ward